lib/db/namespaced.go

```diff
@@ -17,16 +17,18 @@ import (
 // NamespacedKV is a simple key-value store using a specific namespace within
 // a leveldb.
 type NamespacedKV struct {
-	db     *Lowlevel
-	prefix []byte
+	db        *Lowlevel
+	prefix    []byte
+	prefixLen int
 }
 
 // NewNamespacedKV returns a new NamespacedKV that lives in the namespace
 // specified by the prefix.
 func NewNamespacedKV(db *Lowlevel, prefix string) *NamespacedKV {
 	return &NamespacedKV{
-		db:     db,
-		prefix: []byte(prefix),
+		db:        db,
+		prefix:    []byte(prefix),
+		prefixLen: len(prefix),
```

Comment:
If this is just an optimization I think we don't need it. `len(prefix)` is already just a read of the length word in the slice header, even though it looks like a function call.